jstests/test_telemetry.js

```diff
@@ -0,0 +1,179 @@
+(function() {
+"use strict";
+
+const telmPath = "/usr/local/percona/telemetry/psmdb";
+const setParameterOpts = {
+    perconaTelemetryGracePeriod: 2,
+    perconaTelemetryScrapeInterval: 5,
+    perconaTelemetryHistoryKeepInterval: 9
+};
+
+var cleanupTelmDir = function() {
+    var files = listFiles(telmPath);
+    files.forEach((file) => {
+        removeFile(file.name)
+    });
+};
+
+var getTelmRawData = function() {
+    var files = listFiles(telmPath);
+    var data = '';
+    files.forEach((file) => {
+        data = data + cat(file.name)
+    });
+    return data;
+};
+
+var getTelmInstanceId = function(conn) {
+    var cmdLineOpts = conn.getDB("admin").runCommand({getCmdLineOpts: 1});
+    var dbPath = cmdLineOpts['parsed']['storage']['dbPath'];
+    var telmId = _readDumpFile(dbPath + "/psmdb_telemetry.data");
+    return telmId[0]['db_instance_id'].str;
+};
+
+var getTelmDataByConn = function(conn) {
+    var id = getTelmInstanceId(conn);
+    var files = listFiles(telmPath);
+    var data = [] ;
+    files.forEach((file) => {
+        if (file.name.includes(id)) {
+            data.push(JSON.parse(cat(file.name)))
+        }
+    });
+    return data;
+};
+
+var telmTestSingle = function() {
+    mkdir(telmPath);
+    cleanupTelmDir();
+
+    var singleTest = MongoRunner.runMongod({
+        setParameter: setParameterOpts
+    });
+
+    //test perconaTelemetryGracePeriod
+    sleep(3000);
+    var telmFileList = listFiles(telmPath);
+    assert.eq(1,telmFileList.length,telmFileList);
+
+    //test telemetry data
+    var jsonTelmData = getTelmDataByConn(singleTest)[0];
+    jsTest.log("Get single-node telemetry");
+    jsTest.log(jsonTelmData);
+
+    assert(jsonTelmData['pro_features'],"pro_features doesn't exist");
+    if ( jsonTelmData['pro_features'].length > 0 ) {
+        assert.eq('mongod-pro',jsonTelmData['source'],jsonTelmData['source']);
```

Comment:
The '-pro' suffix wil appear in the 'pillar_version' metric, not in the 'source'.